proxy/Main.cc

```diff
@@ -1793,9 +1795,14 @@ main(int /* argc ATS_UNUSED */, const char **argv)
   ink_dns_init(makeModuleVersion(HOSTDB_MODULE_MAJOR_VERSION, HOSTDB_MODULE_MINOR_VERSION, PRIVATE_MODULE_HEADER));
   ink_split_dns_init(makeModuleVersion(1, 0, PRIVATE_MODULE_HEADER));
 
+  naVecMutex             = new_ProxyMutex();
+  started_et_net_threads = 0;
+
   // Do the inits for NetProcessors that use ET_NET threads. MUST be before starting those threads.
   netProcessor.init();
-  init_HttpProxyServer();
+  prep_HttpProxyServer();
+
+  eventProcessor.schedule_spawn(&init_HttpProxyServer, ET_NET);
```

Comment:
Have you considered not doing `schedule_spawn` if `num_accept_threads > 0`? In fact, you could set the condition variable if `num_accept_threads > 0` and then always wait on it.